src/Build/Evaluation/ItemSpec.cs

```diff
@@ -154,37 +284,59 @@ private List<ItemFragment> BuildItemFragments(IElementLocation itemSpecLocation,
 
             isItemListExpression = true;
 
-            return new ItemExpressionFragment<P, I>(capture, expression, this, projectDirectory);
+            return new ItemExpressionFragment(capture, expression, this, projectDirectory);
         }
 
         /// <summary>
-        /// Return true if the given <paramref name="item"/> matches this itemspec
+        ///     Return true if the given <paramref name="item" /> matches this itemspec
         /// </summary>
         /// <param name="item">The item to attempt to find a match for.</param>
         public bool MatchesItem(I item)
         {
             // Avoid unnecessary LINQ/Func/Enumerator allocations on this path, this is called a lot
 
-            string evaluatedInclude = item.EvaluatedInclude;
-            foreach (ItemFragment fragment in Fragments)
+            var evaluatedInclude = item.EvaluatedInclude;
+            foreach (var fragment in Fragments)
```

Comment:
Personal preference. I prefer suggestive names instead of random names with types. Using `var` forces me to use meaningful names.

Reply:
I like meaningful names, but I also like it when I can click on something, hit f12, and find relevant code. I've had a problem before where something was defined vaguely, and the compiler couldn't figure out precisely what type it was, so it sent me somewhere useless instead of sending me to the relevant code.

Reply:
I'd rather we fix these things by adding a repo wide formatting and code style config file (e.g. R# not only formats code, but also fixes C# rules, like `var - > type`) that gets applied on every PR. Otherwise we get blocked in "spaces vs tabs" discussions.

Reply:
https://github.com/microsoft/msbuild/issues/5033

Reply:
I actually think var vs. type is functional, not just stylistic, though.